### PR TITLE
[WIP] (MODULES-4037) Used garethr/puppet-module-skeleton

### PR DESCRIPTION
**Pull request by HAIL9000**
Prior to this commit, 'puppet module generate' was using an
outdated module skeleton. Update it to use the skeleton used by
garethr's tool instead since it's more up to date and is already
more widely used by module developers. Plus this will create more
consistency for module authors since we'll no longer have two
different conflicting tools.

### CODE REVIEW

**Comment by HAIL9000** [comment]
Travis is failing because the commit message references a MODULES ticket. I think this was just an oversight in the way the check was written, but if someone feels it's necessary we could migrate MODULES-4037 to a PUP ticket and reference that instead.

File: lib/puppet/module_tool/skeleton/templates/generator/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+bundler_args: --without development system_tests
+before_install: rm Gemfile.lock || true
+rvm:
+  - 1.8.7
+  - 1.9.3
+  - 2.0.0
+  - 2.1.0
+script: bundle exec rake test
+env:
+  - PUPPET_GEM_VERSION="~> 2.7.0"
```
Comment (bmjen) [inline lib/puppet/module_tool/skeleton/templates/generator/.travis.yml]: #sadness

Comment (HAIL9000) [inline lib/puppet/module_tool/skeleton/templates/generator/.travis.yml]: Oh gosh :x Clearly this was pretty out of date

**Comment by HAIL9000** [comment]
Please don't merge this PR yet. It was filed prematurely and we're in the middle of more discussion on whether or not this is the right approach

**Comment by HAIL9000** [comment]
We've concluded this isn't the write approach, planning to rescope and ticket this work. As such, closing this pull request.

**Pull request closed without merging**